client/mgcp/driver/src/main/java/org/restcomm/media/client/mgcp/stack/JainMgcpStackProviderImpl.java

```diff
@@ -454,6 +456,7 @@ public void activate() {
          */
         private void shutdown() {
             this.active = false;
+            this.interrupt();
```

Comment:
I think it will be enough to set `active=false` and let the thread elegantly terminate by itself.